addons/survey/models/survey_question.py

```diff
@@ -681,17 +681,20 @@ class SurveyQuestionAnswer(models.Model):
     is_correct = fields.Boolean('Correct')
     answer_score = fields.Float('Score', help="A positive score indicates a correct choice; a negative or null score indicates a wrong answer")
 
-    @api.depends('value', 'question_id.title')
+    @api.depends('value', 'question_id.question_type', 'question_id.title', 'matrix_question_id')
     def _compute_display_name(self):
         """Render an answer name as "Question title : Answer value" making sure it is not too long.
 
         This implementation makes sure we have at least 30 characters for the question title,
         then we elide it, leaving the rest of the space for the answer.
         """
         for answer in self:
-            # _origin (or fallback title) is (likely temporarily) needed to support survey snapshot
-            # during onchange for a deleted answer used as trigger in another question.
-            title = answer._origin.question_id.title
+            # _origin is (likely temporarily) needed to support survey snapshot during onchange
+            # for a deleted answer used as trigger in another question.
+            if answer._origin.matrix_question_id or answer._origin.question_id.question_type == 'matrix':
```

Comment:
But why using origin ? It is a compute, it should be based on question_id / matrix_question_id ...

Reply:
We broke something with the normal flow of `onchange` in our survey form view (and questions modal) where we automatically save when we open/close the record (we're pushing the limits here; which we discussed with the JS and python framework). We didn't get to the absolute bottom of it and merged this like this a few weeks ago because it didn't seem worth it when comparing to the 'quick hack'.

Basically, if I remember correctly, it seems like an `onchange` is triggered but it doesn't provide all the data necessary for the compute (because the record hasn't changed?) and so, while `_origin` is correct and contains the right reference(s), `answer.question_id` is `False` during the onchange's snapshot creation, and this causes crashes during our integration test (`survey_tour_test_survey_form_triggers`).
tldr; If we remove `_origin`, we do not get the `question_id` that does exist but `False`. I just confirmed that this is still the case (with all the `api.depends` specified here).

Reply:
It is an issue, we cannot have code that relies on _origin (which is an orm internal trick). If we are pushing the fw too far, maybe we should come back to standard fw. There is no reason a compute should be that hackish ...

Reply:
OK!
I'll discuss this with RCO then, as it is related to the recent refactoring of JS relational model and `onchange2`. 
We didn't think it was worth pursing further (after already several hours) because there are already many instances of using `_origin` in computes and we have good test coverage in case something changes.